java/flight/src/main/protobuf/flight.proto

```diff
@@ -0,0 +1,260 @@
+   * a Pickled Python operation that will be executed. In those cases, the descriptor will not be
+   * previously available within the list of available streams provided by ListFlights but will be
+   * available for consumption for the duration defined by the specific flight service.
+   */  
+  rpc GetFlightInfo(FlightDescriptor) returns (FlightGetInfo) {}
+  
+  /*
+   * Retrieve a single stream associated with a particular descriptor associated with the referenced
+   * ticket. A Flight can be composed of one or more streams where each stream can be retrieved using
+   * a separate opaque ticket that the flight service uses for managing a collection of streams.
+   */
+  rpc DoGet(Ticket) returns (stream FlightData) {}
+
+  /*
+   * Push a stream to the flight service using associated with a particular flight stream. This allows
+   * a client of a flight service to upload a stream of data. Depending on the particular flight
+   * service, a client consumer could be allowed to upload a single stream per descriptor or an
+   * unlimited number. (In the latter, the service might implement a 'seal' action that can be applied
+   * to a descriptor once all streams are uploaded.
+   */  
+  rpc DoPut(stream FlightData) returns (PutResult) {}
+  
+  /*
+   * Flight services can support an arbitrary number of simple actions in addition to the possible
+   * ListFlights, GetFlightInfo, DoGet, DoPut operations that are potentially available. DoAction
+   * allows a flight client to do a specific action against a flight service. An action includes
+   * opaque request and response objects that are specific to the type action being undertaken.
+   */  
+  rpc DoAction(Action) returns (stream Result) {}
+  
+  /*
+   * A flight service exposes all of the available action types that it has along with descriptions.
+   * This allows different flight consumers to understand the capabilities of the flight servic
+   */
+  rpc ListActions(Empty) returns (stream ActionType) {}
+  
+}
+
+/*
+ * The request that a client provides to a server on handshake.
+ */
+message HandshakeRequest {
+
+  /*
+   * A defined protocol version
+   */
+  uint64 protocol_version = 1;
+  Credential credential = 2;
+  bytes payload = 3;
+}
+
+message HandshakeResponse {
+  uint64 protocol_version = 1;
+  bytes payload = 2;
+}
+
+message Credential {
+  enum CredentialType {
+    UNKNOWN = 0;
+    BASIC = 1;
+    OPAQUE = 2;
+  }
+  
+  CredentialType type = 1;
+  string username = 2;
+  string password = 3;
+  bytes opaque_value = 4;
+}
+
+message Empty {}
+
+/*
+ * Describes an available action, including both the name used for execution along with a short
+ * description of the purpose of the action.
+ */
+message ActionType {
+  string type = 1;
+  string description = 2;
+}
+
+/*
+ * A service specific expression that can be used to return a limited the set of available Arrow
+ * Flight streams.
+ */
+message Criteria {
+  bytes expression = 1;
```

Comment:
One question here might be the structure of Criteria, e.g. if there are different kinds of criteria that may be supported. I think it's fine to be opaque for now